tests/functional/test_attention.py

```diff
@@ -0,0 +1,120 @@
+"""Tests for the jax_layers/functional/attention.py implementations."""
+
+from unittest.mock import MagicMock
+
+import jax
+import jax.numpy as jnp
+
+from jax_layers.functional import dot_product_attention
+
+
+def test_dot_product_attention():
+    """Test that the dot_product_attention function works with different implementations."""
+    batch_size = 2
+    seq_len = 16
+    num_heads = 4
+    head_dim = 8
+
+    key = jax.random.PRNGKey(0)
+    key1, key2, key3 = jax.random.split(key, 3)
+
+    query = jax.random.normal(key1, (batch_size, seq_len, num_heads, head_dim))
+    key_tensor = jax.random.normal(key2, (batch_size, seq_len, num_heads, head_dim))
+    value = jax.random.normal(key3, (batch_size, seq_len, num_heads, head_dim))
+
+    # Execute using default (XLA) implementation
+    output_default = dot_product_attention(query, key_tensor, value)
+    output_xla = dot_product_attention(query, key_tensor, value, implementation="xla")
```

Comment:
Instead of the current cross-testing approach,
we could check the arguments by mocking `jax.nn.dot_product_attention`. (same as https://github.com/KennethanCeyer/jax-layers/blob/2c48f62ae6ac0c3882906978c16da0acb3fb73ae/tests/functional/test_attention.py#L50-L61)

However, the previous test setup is not a fully isolated unittest using only patches.
Rather, it seems an integration test that allows some parts of jax.nn to operate.
Thus, I added this setup to ensure reliability through cross-testing to keep the original test convention/purpose.

That said, based on further comment,
we can modify the approach to control `jax.nn` via patches and verify only whether `functional/attention.py` is being called.